Update to rhel9

praveenkumar opened the pull request.

praveenkumar committed:
Update the source image and image family name to rhel9

praveenkumar committed:
IMAGES.md: update the source image is 'rhel9' instead 'rhel8'

praveenkumar committed:
IMAGES.md: update the onboarding link from mojo to source

praveenkumar committed:
IMAGES.md: update steps for getting GCP's application default credentials

praveenkumar committed:
provision: Remove 'libvirt-devel' package

Looks like from RHEL-9 `libvirt-devel` package is not released as part
of appstream repo and only part of codeready-builder-for-rhel-9-<arch>-rpms__9 repo.
We anyway not using this package so better to remove it.

- https://errata.devel.redhat.com/release_engineering/show_released_build/2527239

praveenkumar committed:
Add delete-network-and-subnet.sh script

We only have create-network-and-subnet.sh but after creating the images
it is good to have an script which delete those resource.

cfergeau commented on IMAGES.md:
```diff
@@ -12,7 +12,10 @@ Images are built with [Packer](https://www.packer.io). Override variables as nec
 
 The source image is `rhel9` with [nested virtualization enabled](https://cloud.google.com/compute/docs/instances/enable-nested-virtualization-vm-instances#restrictions).
 
+To use packer with GCP first need to set [application default credentials](https://cloud.google.com/docs/authentication/application-default-credentials)
```
"To use packer with GCP, you first need to set ..."

praveenkumar replied:
@cfergeau updated.

cfergeau commented on openshift4-libvirt.json:
```diff
@@ -1,7 +1,7 @@
   "variables": {
-    "image_family": "openshift4-libvirt",
-    "source_image_family": "openshift4-libvirt-source",
+    "image_family": "openshift4-libvirt-rhel9",
```
Does this need to be changed in `create-gcp-instance.sh`: `  --image-family openshift4-libvirt \` ?

It would make sense to either rename or add new  files which have the same name as the image `openshift4-libvirt-rhel9.json`

praveenkumar replied:
@cfergeau `create-gcp-instance.sh` is something I can change but this file is not used for creating and uploading the GCP image but more like testing part and it does make sense to update this also with correct image-family name.

praveenkumar replied:
updated

praveenkumar committed:
provision: Use modular libvirt approach

RHEL-9 by default have modular libvirt approach to start the required
service and till now we are trying to use mix and match to start
libvirt-tcp socket using old way so this pr should fix it.

cfergeau reviewed (commented):
Looks good to me, still wondering about:
« It would make sense to either rename or add new files which have the same name as the image openshift4-libvirt-rhel9.json »

praveenkumar commented:
> It would make sense to either rename or add new files which have the same name as the image openshift4-libvirt-rhel9.json 

@cfergeau as of now we keep file name same and have rhel-8 branch in case someone want to generate rhel8 image then checkout that branch and use same instruction otherwise we need to change the instruction in the README for each rhel version branch.

cfergeau commented:
> > It would make sense to either rename or add new files which have the same name as the image openshift4-libvirt-rhel9.json
> 
> @cfergeau as of now we keep file name same and have rhel-8 branch in case someone want to generate rhel8 image then checkout that branch and use same instruction otherwise we need to change the instruction in the README for each rhel version branch.

rhel version change won't happen too often :)

praveenkumar commented:
> rhel version change won't happen too often :)

@cfergeau  yes but then we already have branch for specific version of rhel, I don't get why the file name need to have rhel version, if it is only for one specific version.

cfergeau commented:
> > rhel version change won't happen too often :)
> 
> @cfergeau yes but then we already have branch for specific version of rhel, I don't get why the file name need to have rhel version, if it is only for one specific version.

file name and image name were in sync before this change, so I asked why they are no longer in sync. And it seems we could support all rhel versions in a single branch actually, since the image name is different anyway.

praveenkumar commented:
> And it seems we could support all rhel versions in a single branch actually, since the image name is different anyway.

@cfergeau no, because the provision script is different changes for RHEL8 and RHEL-9 because of libvirt modular nature.
